src/continuous_agg.c

```diff
@@ -577,7 +577,15 @@ ts_continuous_agg_drop_hypertable_callback(int32 hypertable_id)
 
 		if (data->raw_hypertable_id == hypertable_id)
 		{
-			continuous_agg_init(&ca, data);
+			/* schema might not exists at this point */
+			Oid nspid = get_namespace_oid(NameStr(data->user_view_schema), true);
+
+			if (OidIsValid(nspid))
```

Comment:
I understand your concern. I tried to make this fix as smaller as possible because we already have a lot of assumption that this functionality can be called during `CASCADE` and expects some of the objects to be deleted: https://github.com/timescale/timescaledb/blob/master/src/continuous_agg.c#L459

Reply:
Well, those assumptions aren't great either, TBH.

I am actually most concerned about this PR introducing an ambiguous state for the `Hypertable` object, in a change below. We've had issues related to such things before where some fields could be set or not set, and we've fixed most of them to always be consistent. This change would move us back towards a situation where objects can again be in different "states".